_posts/2024-04-30-A-comprehensive-guide-to-setup-and-connect-self-managed-dashboards-with-Amazon-OpenSearch-domain.md

```diff
@@ -0,0 +1,276 @@
+                },
+                {
+                    "name": "dash-9200-tcp",
+                    "containerPort": 9200,
+                    "hostPort": 9200,
+                    "protocol": "tcp",
+                    "appProtocol": "http"
+                }
+            ],
+            "essential": true,
+            "environment": [
+                {
+                    "name": "OPENSEARCH_USERNAME",
+                    "value": "xxx"
+                },
+                {
+                    "name": "OPENSEARCH_HOSTS",
+                    "value": "https://success-2-ce6hkjt5gh.ap-south-1.es.amazonaws.com"
+                },
+                {
+                    "name": "OPENSEARCH_PASSWORD",
+                    "value": "xxxx"
+                }
+            ],
+...
+}
+```
+
+4. Create a service using the previously created task within the identical VPC and subnet where the managed service OpenSearch domain is operating.
+5. Access the self-managed dashboards by hitting the public endpoint of the running task in ECS Fargate. By doing so, you can conveniently view and interact with all the saved objects in accordance with the Fine-Grained Access Control settings.
+
+## Steps to spin up a self-managed dashboards in EC2 hosted container
+1. Deploy an EC2 instance in the same VPC and subnet as the managed OpenSearch service domain.
+2. Set up Docker/Kubernetes and its dependencies on the instance.
+3. Utilize the below `docker-compose.yml` file to launch a self-managed dashboards container. After the container is running, you can easily access and interact with all the saved objects.
+```yml
+version: '3'
+services:
+  opensearch-dashboards:
+    image: opensearchproject/opensearch-dashboards:2.5.0
+    container_name: opensearch-dashboards
+    ports:
+      - 5601:5601
+    expose:
+      - "5601"
+    environment:
+      OPENSEARCH_HOSTS: '["https://success-2-ce6hkjt5gh.ap-south-1.es.amazonaws.com"]'
+      OPENSEARCH_USERNAME: 'xxx'
+      OPENSEARCH_PASSWORD: 'xxxx'
+    networks:
+      - opensearch-net
+networks:
+  opensearch-net:
+```
+4. To enable TLS, add the attributes mentioned in this [link](https://opensearch.org/docs/latest/install-and-configure/install-dashboards/tls/) as environment variables.
+
+> [!TIP]
+> One notable advantage of setting up a self-managed dashboards is that when it is deployed on `AWS ECS Fargate`, it generates a public IP. This allows the self-managed dashboards to be accessed over the internet without the need for setting up a reverse proxy. As a result, the OpenSearch domains will be within the VPC and self-managed dashboards will be available in public, enabling seamless connectivity and eliminating the complexity of configuring additional infrastructure components. This simplifies the setup process and provides convenient access to the dashboards from anywhere on the internet without compromising security or requiring additional network configurations.
```

Comment:
Is `AWS ECS Fargate` intentionally in code font, or should it be "Amazon ECS on AWS Fargate" (https://docs.aws.amazon.com/AmazonECS/latest/developerguide/AWS_Fargate.html)?

Reply:
This is just a generic term and if "Amazon ECS on AWS Fargate" is the right set of words recommended, lets stick to this.